tests/experimental/test_pfs.py

```diff
@@ -773,3 +734,138 @@ def test_mount():
     Path("mount_d/file.txt").touch()
     with pytest.raises(RuntimeError, match="must be empty to mount"):
         client.mount("mount_d")
+
+
+@pytest.fixture(name="repo")
+def _repo_fixture(request) -> str:
+    """Create a repository name from the test function name."""
+    return request.node.nodeid.replace("/", "-").replace(":", "-").replace(".py", "")
+
+
+@pytest.fixture(name="client")
+def _client_fixture(repo):
+    client = Client()
+    client.delete_repo(repo, force=True)
+    client.create_repo(repo, "test repo for python_pachyderm")
+    yield client
+    client.delete_repo(repo, force=True)
+
+
+class TestPFSFile:
+    @staticmethod
+    def test_get_large_file(client: Client, repo: str):
+        """Test that a large file (requires >1 gRPC message to stream)
+        is successfully streamed in it's entirety
+        """
+        # Arrange
+        data = os.urandom(int(MAX_RECEIVE_MESSAGE_SIZE * 1.1))
+        pfs_file = "/large_file.dat"
+
+        with client.commit(repo, "large_file_commit") as commit:
+            client.put_file_bytes(commit, pfs_file, data)
+
+        # Act
+        with client.get_file(commit, pfs_file) as file:
+            assert len(file._buffer) < len(data), (
+                "PFSFile initialization streams the first message. "
+                "This asserts that the test file must be streamed over multiple messages. "
+            )
+            streamed_data = file.read()
+
+        # Assert
+        assert streamed_data[:10] == data[:10]
+        assert streamed_data[-10:] == data[-10:]
+        assert len(streamed_data) == len(data)
+
+    @staticmethod
+    def test_buffered_data(client: Client, repo: str):
+        """Test that data gets buffered as expected."""
+        # Arrange
+        stream_items = [b"a", b"bc", b"def", b"ghij", b"klmno"]
+        stream = (BytesValue(value=item) for item in stream_items)
```

Comment:
This parenthesis syntax works?

Reply:
Yep, it's the comprehension-syntax for iterators, basically just a lazy list comprehension

Reply:
Sick. I didn't know generators had their own comprehension too.